src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -466,15 +467,35 @@ private ArgumentListSyntax Transform( ArgumentListSyntax argList )
 	private TypeArgumentListSyntax Transform( TypeArgumentListSyntax typeArgList )
 		=> typeArgList.WithArguments( TransformTypes( typeArgList.Arguments ) );
 
+	private ParameterListSyntax Transform( ParameterListSyntax paramList )
+		=> paramList.WithParameters( TransformParams( paramList.Parameters ) );
+
 	private ArgumentSyntax? Transform( ArgumentSyntax argument ) {
 		if( Model.GetTypeInfo( argument.Expression ).Type?.Name == "CancellationToken" ) {
 			return null;
 		}
 		return argument.WithExpression( Transform( argument.Expression ) );
 	}
 
-	private ParameterSyntax Transform( ParameterSyntax parameter )
-		=> parameter.WithIdentifier(RemoveAsyncSuffix( parameter.Identifier, optional: true ) );
+	private SeparatedSyntaxList<ParameterSyntax> TransformParams( SeparatedSyntaxList<ParameterSyntax> paramSynts ) {
+		for( int i = 0; i < paramSynts.Count; i++ ) {
+			ParameterSyntax? newParamSynt = Transform( paramSynts[i] );
+			if( newParamSynt is not null ) {
+				paramSynts = paramSynts.Replace( paramSynts[i], newParamSynt );
+			} else {
+				paramSynts = paramSynts.RemoveAt( i );

```

Comment:
I think this might not work when the CancellationToken is in the middle? because we are modifying the size of `paramSynts` while we are iterating over it, so if we remove the ith element the next run of the loop will be `i+1` and miss an element of `paramSynts`.

There are a few solutions. One is to do `i--` here.

Another is to build a new `List<ParameterSyntax>` to pass into `SyntaxFactory.SeparatedList`. Instead of calling `Replace` above you would `Add` to the list, and instead of `RemoveAt` you would just not do anything.

One thing to think about is how (I think) it's probably a not-uncommon case for the parameter list to not be modified. The ideal thing in a situation like that would be if we ultimately did `return paramSynts` (on the original/unmodified value). To make that happen you could lazily allocate that `List<ParameterSyntax>` (make it null to start, only allocate it with `new` once you find something that `newParamSynt != paramSynts[i]` for the first time, but then you also have to `Array.Copy` all the things from 0 to i-1 to "catch up"). But since this is a parameter list that only happens once per `[GenerateSync]` I probably wouldn't worry about it (if it was argument lists instead I might -- there could be many argument lists per `[GenerateSync]` and its probably much more common for them to go unmodified.)

Reply:
Can you add a test for

```cs
[GenerateSync]
Task FooAsync( int w, int x, CancellationToken c, int y, int z );
```
?

Reply:
I tried passing into SyntaxFactory.SeparatedList first but there was no way to preserve the comma whitespace and I didn't want the params to get squished together, if that makes sense

Reply:
Ok, I used the i--; fix and added the test but with another adjacent CancellationToken because the test you proposed was still working since it just skipped removing int y haha